File: test-suite-hazelcast-java/src/test/java/io/micronaut/cache/HazelcastAdditionalSettings.java
package io.micronaut.cache;

import io.micronaut.cache.hazelcast.HazelcastClientConfiguration;
import io.micronaut.context.event.BeanCreatedEvent;
import io.micronaut.context.event.BeanCreatedEventListener;
import io.micronaut.core.annotation.NonNull;
import jakarta.inject.Singleton;

// tag::clazz[]
@Singleton
public class HazelcastAdditionalSettings implements BeanCreatedEventListener<HazelcastClientConfiguration> {

    @Override
    public HazelcastClientConfiguration onCreated(@NonNull BeanCreatedEvent<HazelcastClientConfiguration> event) {
        HazelcastClientConfiguration configuration = event.getBean();
        // Set anything on the configuration
        configuration.setClusterName("dev");

        return configuration;
    }
}
// end::clazz[]

File: test-suite-hazelcast-python/src/test/python/micronaut/cache/docs/HazelcastAdditionalSettings.py
from jakarta.inject import Singleton
from micronaut.cache.hazelcast import HazelcastClientConfiguration
from micronaut.context.event import BeanCreatedEvent, BeanCreatedEventListener


# tag::clazz[]
@Singleton
class HazelcastAdditionalSettings(BeanCreatedEventListener[HazelcastClientConfiguration]):

    def onCreated(self, event: BeanCreatedEvent[HazelcastClientConfiguration]) -> HazelcastClientConfiguration:
        configuration = event.getBean()
        # Set anything on the configuration
        configuration.setClusterName("dev")

        return configuration
# end::clazz[]
